examples/simulate_fight_scenario.py

```diff
@@ -0,0 +1,111 @@
+
+    async def on_step(self, iteration):
+        # prepare the level
+        if not self.control_requested:
+            # we need this one for `self.enemy_units` to "see" all units
+            await self._client.debug_show_map()
+            # this one will allow us to do something like: `self.enemy_units.first.attack(self.townhalls.first)`
+            await self._client.debug_control_enemy()
+            logger.info("control requested")
+            # await self.chat_send("control requested")
+            self.control_requested = True
+
+        if self.control_requested and self.enemy_units and not self.control_received:
+            # prepare my side
+            me = 1
+            cc = self.townhalls.first
+            p = cc.position.towards(self.game_info.map_center, 3)
+            # create supply
+            await self._client.debug_create_unit([[UnitTypeId.SUPPLYDEPOT, 1, p, me]])
+            # destroy command center
+            await self._client.debug_kill_unit([cc.tag])
+            # destroy all workers
+            for w in self.units(UnitTypeId.SCV):
+                await self._client.debug_kill_unit([w.tag])
+            # create marines
+            await self._client.debug_create_unit([[UnitTypeId.MARINE, 4, p, me]])
+
+            # prepare opponent side
+            pc = 2
+            cc = self.enemy_structures.first
+            p = cc.position.towards(self.game_info.map_center, 3)
+            # create supply
+            await self._client.debug_create_unit([[UnitTypeId.SUPPLYDEPOT, 1, p, pc]])
+            # destroy command center
+            await self._client.debug_kill_unit([cc.tag])
+            # destroy all workers
+            for w in self.enemy_units(UnitTypeId.SCV):
+                await self._client.debug_kill_unit([w.tag])
+            # create marines
+            await self._client.debug_create_unit([[UnitTypeId.MARINE, 4, p, pc]])
+            logger.info("control received")
+            # await self.chat_send("control received")
+            self.control_received = True
+
+        # to speedup, we are going damage both supplies
+        if not self.supplies_been_damaged and self.structures(UnitTypeId.SUPPLYDEPOT) and self.enemy_structures(UnitTypeId.SUPPLYDEPOT):
+            for s in self.structures:
+                await self._client.debug_set_unit_value([s.tag], 2, 100)
+            for s in self.enemy_structures:
+                await self._client.debug_set_unit_value([s.tag], 2, 100)
+            logger.info("supplies damaged")
+            # await self.chat_send("supplies damaged")
+            self.supplies_been_damaged = True
+
+        # note: we should wait till workers will be destroyed
+        if not self.fight_started and self.control_received and self.enemy_units and not self.enemy_units(UnitTypeId.SCV) and not self.units(UnitTypeId.SCV):
+            # start fight
+            for u in self.enemy_units:
+                u.attack(self.structures.first.position)
```

Comment:
Yes, it does, we are just simulating the world start position, then it will be computers AI decisions, e.g. he can retrieve or continue attack or do something else like if we play against him usual full length game

As you have mentioned we can start game with two bots and control everything, but then it will be not the same - e.g. with this approach I can test my piece of logic against medium and hard computer

PS: we can prevent computer will by just sending units into attack always without checks, but then again it will be not the same as in usual game